addons/base_import/models/base_import.py

```diff
@@ -845,6 +848,68 @@ def _import_image_by_url(self, url, session, field, line_number):
                 'error': e
             })
 
+    @api.model
+    def _check_external_ids(self, data, import_fields):
```

Comment:
very big method may be you can split in 2 or 3 ?

Reply:
Everything is very interlaced, it will be even harder to read. There is also no duplicated code, so no reason to split it.
But I can definitely add some comments to help following the data flow

Reply:
yes thanks it's will be great ! :pray: